### ISSUE TITLE
Loads of requests!

### ISSUE THREAD

**Issue by jasonwragg** [ljmerza/frigate_plate_recognizer#2]
Not sure what I did wrong but I burned through my 2500 requests in a few hours.

**Comment by ljmerza** [comment]
Turn on debug mode and take a look at how many times it's finding a matching object to plate recognize (ie car, bus, motorcycle)

**Comment by Heisenberg2980** [comment]
Just installed it and might be having the same issue, as the same image is being sent to recognizer every minute after starting the container (I stopped it after some minutes). Looking at the logs it seems an error is produced every minute and when the process restarts it sends the same image to recognizer. The image being sent is the last event of that camera which happened 7 hours ago, so it seems just after starting the container sends the image from the latest event to recognizer (obviously the event happened before starting the container), not sure if that is by design but avoiding this could fix this issue (at least the part of the process sending one image per minute).

This is the log:

```
frigate_plate_recognizer  | Process Process-1:
frigate_plate_recognizer  | Traceback (most recent call last):
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
frigate_plate_recognizer  |     self.run()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/multiprocessing/process.py", line 108, in run
frigate_plate_recognizer  |     self._target(*self._args, **self._kwargs)
frigate_plate_recognizer  |   File "//./index.py", line 213, in run_mqtt_client
frigate_plate_recognizer  |     client.loop_forever()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1756, in loop_forever
frigate_plate_recognizer  |     rc = self._loop(timeout)
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1164, in _loop
frigate_plate_recognizer  |     rc = self.loop_read()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1556, in loop_read
frigate_plate_recognizer  |     rc = self._packet_read()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2439, in _packet_read
frigate_plate_recognizer  |     rc = self._packet_handle()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3033, in _packet_handle
frigate_plate_recognizer  |     return self._handle_publish()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3327, in _handle_publish
frigate_plate_recognizer  |     self._handle_on_message(message)
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3570, in _handle_on_message
frigate_plate_recognizer  |     on_message(self, self._userdata, message)
frigate_plate_recognizer  |   File "//./index.py", line 160, in on_message
frigate_plate_recognizer  |     cursor.execute("""
frigate_plate_recognizer  | sqlite3.IntegrityError: UNIQUE constraint failed: plates.frigate_event
frigate_plate_recognizer  | 2023-09-09 20:47:18,088 - __main__ - INFO - Time: 2023-09-09 20:47:18.087
frigate_plate_recognizer  | 2023-09-09 20:47:18,088 - __main__ - INFO - Python Version: 3.9.18 (main, Aug 26 2023, 01:24:18) 
frigate_plate_recognizer  | [GCC 12.2.0]
frigate_plate_recognizer  | 2023-09-09 20:47:18,088 - __main__ - INFO - Frigate Plate Recognizer Version: 1.2.1
frigate_plate_recognizer  | 2023-09-09 20:47:18,089 - __main__ - INFO - Starting MQTT client. Connecting to: 192.168.0.253
frigate_plate_recognizer  | 2023-09-09 20:47:18,090 - __main__ - INFO - MQTT Connected
```

**Comment by ljmerza** [comment]
added a check in v1.2.2 to make sure we dont process the same event. should be able to just pull latest

**Comment by Heisenberg2980** [comment]
Thanks for the quick update, it looks like both issues are now fixed as the process is not sending the previous event anymore and there isn´t any error in the logs even after a new detection is processed and sent to recognizer

**Comment by jasonwragg** [comment]
I'm still getting a lot of requests with the new version.

`2023-09-10 14:00:13,326 - __main__ - INFO - No plate number found for event 1694343100.433778-mb09ei
2023-09-10 14:00:13,887 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694345415.80259-p0m78e', 'camera': 'front_garden', 'frame_time': 1694350753.464871, 'snapshot_time': 1694347600.131993, 'label': 'car', 'sub_label': None, 'top_score': 0.74609375, 'false_posit>
2023-09-10 14:00:13,896 - __main__ - DEBUG - Getting image for event: 1694345415.80259-p0m78e
2023-09-10 14:00:13,897 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694345415.80259-p0m78e/snapshot.jpg
2023-09-10 14:00:14,892 - __main__ - DEBUG - response: {'processing_time': 75.22, 'results': [], 'filename': '1300_sPTMQ_upload.jpg', 'version': 1, 'camera_id': None, 'timestamp': '2023-09-10T13:00:14.745242Z'}
2023-09-10 14:00:14,892 - __main__ - INFO - No plate number found for event 1694345415.80259-p0m78e
2023-09-10 14:00:14,893 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694343103.981412-dnai3l', 'camera': 'front_garden', 'frame_time': 1694350753.845286, 'snapshot_time': 1694347610.13847, 'label': 'car', 'sub_label': None, 'top_score': 0.84375, 'false_positive>
2023-09-10 14:00:14,894 - __main__ - DEBUG - Getting image for event: 1694343103.981412-dnai3l
2023-09-10 14:00:14,894 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694343103.981412-dnai3l/snapshot.jpg
2023-09-10 14:00:15,615 - __main__ - DEBUG - response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-10 14:00:15,616 - __main__ - ERROR - Failed to get plate number. Response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-10 14:00:15,616 - __main__ - INFO - No plate number found for event 1694343103.981412-dnai3l
2023-09-10 14:00:17,553 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694332866.546505-gxgkuk', 'camera': 'front_garden', 'frame_time': 1694350756.895901, 'snapshot_time': 1694343102.58397, 'label': 'car', 'sub_label': None, 'top_score': 0.84375, 'false_positive>
2023-09-10 14:00:17,554 - __main__ - DEBUG - Getting image for event: 1694332866.546505-gxgkuk
2023-09-10 14:00:17,554 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694332866.546505-gxgkuk/snapshot.jpg
2023-09-10 14:00:18,693 - __main__ - DEBUG - response: {'processing_time': 57.983, 'results': [], 'filename': '1300_s31Vh_upload.jpg', 'version': 1, 'camera_id': None, 'timestamp': '2023-09-10T13:00:18.563746Z'}
2023-09-10 14:00:18,694 - __main__ - INFO - No plate number found for event 1694332866.546505-gxgkuk
2023-09-10 14:00:21,349 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694350816.523305-wh6ga0', 'camera': 'living_room', 'frame_time': 1694350816.523305, 'snapshot_time': 0.0, 'label': 'dog', 'sub_label': None, 'top_score': 0.0, 'false_positive': True, 'start_ti>
2023-09-10 14:00:21,350 - __main__ - DEBUG - Skipping event: 1694350816.523305-wh6ga0 because it is from the wrong camera: living_room
2023-09-10 14:00:24,903 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694343100.433778-mb09ei', 'camera': 'front_garden', 'frame_time': 1694350812.233094, 'snapshot_time': 1694350690.494648, 'label': 'car', 'sub_label': None, 'top_score': 0.84375, 'false_positiv>
2023-09-10 14:00:24,904 - __main__ - DEBUG - Getting image for event: 1694343100.433778-mb09ei
2023-09-10 14:00:24,904 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694343100.433778-mb09ei/snapshot.jpg
2023-09-10 14:00:25,687 - __main__ - DEBUG - response: {'processing_time': 47.834, 'results': [{'box': {'xmin': 56, 'ymin': 184, 'xmax': 117, 'ymax': 199}, 'plate': 'wn5wgn', 'region': {'code': 'gb', 'score': 0.69}, 'score': 0.643, 'candidates': [{'score': 0.643, 'plate'>
2023-09-10 14:00:25,688 - __main__ - INFO - Score is below minimum: 0.643
2023-09-10 14:00:26,549 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694350816.523305-wh6ga0', 'camera': 'living_room', 'frame_time': 1694350821.069491, 'snapshot_time': 1694350821.069491, 'label': 'dog', 'sub_label': None, 'top_score': 0.740234375, 'false_posi>
2023-09-10 14:00:26,550 - __main__ - DEBUG - Skipping event: 1694350816.523305-wh6ga0 because it is from the wrong camera: living_room
2023-09-10 14:00:29,992 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694343100.433778-mb09ei', 'camera': 'front_garden', 'frame_time': 1694350824.476555, 'snapshot_time': 1694350824.476555, 'label': 'car', 'sub_label': None, 'top_score': 0.84375, 'false_positiv>
2023-09-10 14:00:29,993 - __main__ - DEBUG - Getting image for event: 1694343100.433778-mb09ei
2023-09-10 14:00:29,993 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694343100.433778-mb09ei/snapshot.jpg
2023-09-10 14:00:30,722 - __main__ - DEBUG - response: {'processing_time': 36.625, 'results': [{'box': {'xmin': 44, 'ymin': 188, 'xmax': 103, 'ymax': 202}, 'plate': 'rn5wgn', 'region': {'code': 'gb', 'score': 0.749}, 'score': 0.746, 'candidates': [{'score': 0.746, 'plate>
2023-09-10 14:00:30,722 - __main__ - INFO - Score is below minimum: 0.746
2023-09-10 14:00:35,450 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694343100.433778-mb09ei', 'camera': 'front_garden', 'frame_time': 1694350829.639445, 'snapshot_time': 1694350826.2469, 'label': 'car', 'sub_label': None, 'top_score': 0.84375, 'false_positive'>
2023-09-10 14:00:35,450 - __main__ - DEBUG - Getting image for event: 1694343100.433778-mb09ei
2023-09-10 14:00:35,450 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694343100.433778-mb09ei/snapshot.jpg
2023-09-10 14:00:36,249 - __main__ - DEBUG - response: {'processing_time': 76.969, 'results': [], 'filename': '1300_5J9Yg_upload.jpg', 'version': 1, 'camera_id': None, 'timestamp': '2023-09-10T13:00:36.100647Z'}
2023-09-10 14:00:36,249 - __main__ - INFO - No plate number found for event 1694343100.433778-mb09ei
2023-09-10 14:00:36,901 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694343100.433778-mb09ei', 'camera': 'front_garden', 'frame_time': 1694350834.832509, 'snapshot_time': 1694350833.667674, 'label': 'car', 'sub_label': None, 'top_score': 0.84375, 'false_positiv>
2023-09-10 14:00:36,902 - __main__ - DEBUG - Getting image for event: 1694343100.433778-mb09ei
2023-09-10 14:00:36,902 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694343100.433778-mb09ei/snapshot.jpg
2023-09-10 14:00:37,701 - __main__ - DEBUG - response: {'processing_time': 52.92, 'results': [], 'filename': '1300_1N9yn_upload.jpg', 'version': 1, 'camera_id': None, 'timestamp': '2023-09-10T13:00:37.575566Z'}
2023-09-10 14:00:37,701 - __main__ - INFO - No plate number found for event 1694343100.433778-mb09ei
2023-09-10 14:00:38,599 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694327459.66822-qbruy8', 'camera': 'liams_room', 'frame_time': 1694350778.071417, 'snapshot_time': 1694327495.414543, 'label': 'person', 'sub_label': None, 'top_score': 0.7890625, 'false_posit>
2023-09-10 14:00:38,599 - __main__ - DEBUG - Skipping event: 1694327459.66822-qbruy8 because it is from the wrong camera: liams_room
2023-09-10 14:00:39,453 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694321866.87967-6m3iaw', 'camera': 'front_garden', 'frame_time': 1694350778.864343, 'snapshot_time': 1694332609.943982, 'label': 'car', 'sub_label': None, 'top_score': 0.84375, 'false_positive>
2023-09-10 14:00:39,454 - __main__ - DEBUG - Getting image for event: 1694321866.87967-6m3iaw
2023-09-10 14:00:39,454 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694321866.87967-6m3iaw/snapshot.jpg
2023-09-10 14:00:40,696 - __main__ - DEBUG - response: {'processing_time': 79.085, 'results': [{'box': {'xmin': 39, 'ymin': 271, 'xmax': 95, 'ymax': 283}, 'plate': 'mh05mcv', 'region': {'code': 'gb', 'score': 0.444}, 'score': 0.648, 'candidates': [{'score': 0.648, 'plate>
2023-09-10 14:00:40,696 - __main__ - INFO - Score is below minimum: 0.648
2023-09-10 14:00:43,141 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694350223.956533-e1l6ma', 'camera': 'landing', 'frame_time': 1694350782.713611, 'snapshot_time': 1694350224.779714, 'label': 'person', 'sub_label': None, 'top_score': 0.8203125, 'false_positiv>
2023-09-10 14:00:43,141 - __main__ - DEBUG - Skipping event: 1694350223.956533-e1l6ma because it is from the wrong camera: landing
2023-09-10 14:00:46,013 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694350097.188611-endwb2', 'camera': 'back_garden', 'frame_time': 1694350785.538376, 'snapshot_time': 1694350355.198635, 'label': 'person', 'sub_label': None, 'top_score': 0.71484375, 'false_po>
2023-09-10 14:00:46,013 - __main__ - DEBUG - Skipping event: 1694350097.188611-endwb2 because it is from the wrong camera: back_garden
2023-09-10 14:00:46,694 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694350840.06234-7szema', 'camera': 'living_room', 'frame_time': 1694350840.06234, 'snapshot_time': 0.0, 'label': 'person', 'sub_label': None, 'top_score': 0.0, 'false_positive': True, 'start_t>
2023-09-10 14:00:46,694 - __main__ - DEBUG - Skipping event: 1694350840.06234-7szema because it is from the wrong camera: living_room
2023-09-10 14:00:47,066 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694343100.433778-mb09ei', 'camera': 'front_garden', 'frame_time': 1694350836.430737, 'snapshot_time': 1694350833.667674, 'label': 'car', 'sub_label': None, 'top_score': 0.84375, 'false_positiv>
2023-09-10 14:00:47,066 - __main__ - DEBUG - Getting image for event: 1694343100.433778-mb09ei
2023-09-10 14:00:47,067 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694343100.433778-mb09ei/snapshot.jpg
2023-09-10 14:00:47,835 - __main__ - DEBUG - response: {'processing_time': 48.865, 'results': [], 'filename': '1300_PqkDQ_upload.jpg', 'version': 1, 'camera_id': None, 'timestamp': '2023-09-10T13:00:47.715267Z'}
2023-09-10 14:00:47,835 - __main__ - INFO - No plate number found for event 1694343100.433778-mb09ei
2023-09-10 14:00:47,836 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694350222.390967-1x89tc', 'camera': 'landing', 'frame_time': 1694350787.36054, 'snapshot_time': 1694350224.161838, 'label': 'person', 'sub_label': None, 'top_score': 0.775390625, 'false_positi>
2023-09-10 14:00:47,836 - __main__ - DEBUG - Skipping event: 1694350222.390967-1x89tc because it is from the wrong camera: landing
2023-09-10 14:00:49,962 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694350816.523305-wh6ga0', 'camera': 'living_room', 'frame_time': 1694350826.303483, 'snapshot_time': 1694350821.693632, 'label': 'dog', 'sub_label': None, 'top_score': 0.84375, 'false_positive>
2023-09-10 14:00:49,963 - __main__ - DEBUG - Skipping event: 1694350816.523305-wh6ga0 because it is from the wrong camera: living_room
2023-09-10 14:00:50,103 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694350816.523305-wh6ga0', 'camera': 'living_room', 'frame_time': 1694350849.51019, 'snapshot_time': 1694350821.693632, 'label': 'dog', 'sub_label': None, 'top_score': 0.84375, 'false_positive'>
2023-09-10 14:00:50,103 - __main__ - DEBUG - Skipping event: 1694350816.523305-wh6ga0 because it is from the wrong camera: living_room
2023-09-10 14:00:50,566 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694350816.523305-wh6ga0', 'camera': 'living_room', 'frame_time': 1694350849.90899, 'snapshot_time': 1694350821.693632, 'label': 'dog', 'sub_label': None, 'top_score': 0.84375, 'false_positive'>
2023-09-10 14:00:50,566 - __main__ - DEBUG - Skipping event: 1694350816.523305-wh6ga0 because it is from the wrong camera: living_room`

**Comment by ljmerza** [comment]
I think there's something wrong with either mqtt or frigate config because it looks like messages are coming in the from mqtt that are dups. Id get something like mqtt explorer and see if dup messages are being sent

**Comment by Heisenberg2980** [comment]
This "loop" issue has reappeared again in 1.3.0, and the process is still sending the same image every minute. It seems the error is now happening when the process tries to publish a message to the return topic.

I think this loop error will cause some people to use a lot of their API calls or even reach the 2500 calls if they don´t realise an error is happening, is there a way to avoid this loop from happening ever again for every potential error in the future? Maybe the process could exit/stop after 3/5 consecutive errors or something like that?

This is the error log:

```
frigate_plate_recognizer  | Process Process-1:
frigate_plate_recognizer  | Traceback (most recent call last):
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
frigate_plate_recognizer  |     self.run()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/multiprocessing/process.py", line 108, in run
frigate_plate_recognizer  |     self._target(*self._args, **self._kwargs)
frigate_plate_recognizer  |   File "//./index.py", line 245, in run_mqtt_client
frigate_plate_recognizer  |     mqtt_client.loop_forever()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1756, in loop_forever
frigate_plate_recognizer  |     rc = self._loop(timeout)
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1164, in _loop
frigate_plate_recognizer  |     rc = self.loop_read()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1556, in loop_read
frigate_plate_recognizer  |     rc = self._packet_read()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2439, in _packet_read
frigate_plate_recognizer  |     rc = self._packet_handle()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3033, in _packet_handle
frigate_plate_recognizer  |     return self._handle_publish()
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3327, in _handle_publish
frigate_plate_recognizer  |     self._handle_on_message(message)
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3570, in _handle_on_message
frigate_plate_recognizer  |     on_message(self, self._userdata, message)
frigate_plate_recognizer  |   File "//./index.py", line 190, in on_message
frigate_plate_recognizer  |     mqtt_client.publish(topic, {
frigate_plate_recognizer  |   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1247, in publish
frigate_plate_recognizer  |     raise TypeError(
frigate_plate_recognizer  | TypeError: payload must be a string, bytearray, int, float or None.
frigate_plate_recognizer  | 2023-09-11 09:45:07,381 - __main__ - INFO - Time: 2023-09-11 09:45:07.381
frigate_plate_recognizer  | 2023-09-11 09:45:07,381 - __main__ - INFO - Python Version: 3.9.18 (main, Sep  7 2023, 13:31:18) 
frigate_plate_recognizer  | [GCC 12.2.0]
frigate_plate_recognizer  | 2023-09-11 09:45:07,381 - __main__ - INFO - Frigate Plate Recognizer Version: 1.3.0
frigate_plate_recognizer  | 2023-09-11 09:45:07,383 - __main__ - INFO - Starting MQTT client. Connecting to: 192.168.0.253
frigate_plate_recognizer  | 2023-09-11 09:45:07,384 - __main__ - INFO - MQTT Connected
```

**Comment by Heisenberg2980** [comment]
Thanks @ljmerza, have you chosen one (or both) of the solutions I was proposing or a different one? just wondering if we should expect the process to automatically stop after certain consecutive failures or to keep restarting if a failure occurs but without sending the picture to plate recognizer (tbh I think I like this second option more)

**Comment by ljmerza** [comment]
I removed the threading to so the program should just crash and stop. The db records need the plate information which is why it comes after the plate recognition call but before the set label and mqtt return code.

**Comment by yehpo** [comment]
@ljmerza Should I rather pull 1.2.9?

**Comment by ljmerza** [comment]
can you post the trace stack?

**Comment by jasonwragg** [comment]
I still get the same image being processed again and again. I presuming I don't have to adjust anything in Frigate? The cars in the pictures are stationary.

**Comment by heisenberg2980** [comment]
@jasonwragg that is strange, are you sure you are running the latest version (1.3.2), did you do a pull to get it? what are your logs showing?

**Comment by jasonwragg** [comment]
Yes I'm used the lastest version.

The follow are processed over and over again.

`2023-09-14 16:44:28,392 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694705821.469712-e1649o/snapshot.jpg
2023-09-14 16:44:29,011 - __main__ - DEBUG - response: {'processing_time': 79.532, 'results': [], 'filename': '1544_6YdHm_upload.jpg', 'version': 1, '>
2023-09-14 16:44:29,011 - __main__ - INFO - No plate number found for event 1694705821.469712-e1649o
2023-09-14 16:44:29,012 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694706255.752426-p9ig97', 'camera': 'living_room', 'frame_time': 169470>
2023-09-14 16:44:29,012 - __main__ - DEBUG - Skipping event: 1694706255.752426-p9ig97 because it is from the wrong camera: living_room
2023-09-14 16:44:29,012 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694706204.7118-wyd02k', 'camera': 'living_room', 'frame_time': 16947062>
2023-09-14 16:44:29,012 - __main__ - DEBUG - Skipping event: 1694706204.7118-wyd02k because it is from the wrong camera: living_room
2023-09-14 16:44:30,888 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694703858.328741-pq26e8', 'camera': 'living_room', 'frame_time': 169470>
2023-09-14 16:44:30,888 - __main__ - DEBUG - Skipping event: 1694703858.328741-pq26e8 because it is from the wrong camera: living_room
2023-09-14 16:44:33,905 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694705985.304977-5fmjmh', 'camera': 'living_room', 'frame_time': 169470>
2023-09-14 16:44:33,909 - __main__ - DEBUG - Skipping event: 1694705985.304977-5fmjmh because it is from the wrong camera: living_room
2023-09-14 16:44:34,155 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694706204.7118-wyd02k', 'camera': 'living_room', 'frame_time': 16947062>
2023-09-14 16:44:34,155 - __main__ - DEBUG - Skipping event: 1694706204.7118-wyd02k because it is from the wrong camera: living_room
2023-09-14 16:44:39,367 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694706204.7118-wyd02k', 'camera': 'living_room', 'frame_time': 16947062>
2023-09-14 16:44:39,367 - __main__ - DEBUG - Skipping event: 1694706204.7118-wyd02k because it is from the wrong camera: living_room
2023-09-14 16:44:44,403 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694696816.207402-fnl86q', 'camera': 'front_garden', 'frame_time': 16947>
2023-09-14 16:44:44,404 - __main__ - DEBUG - Getting image for event: 1694696816.207402-fnl86q
2023-09-14 16:44:44,404 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694696816.207402-fnl86q/snapshot.jpg
2023-09-14 16:44:45,454 - __main__ - DEBUG - response: {'processing_time': 79.418, 'results': [], 'filename': '1544_AVlbQ_upload.jpg', 'version': 1, '>
2023-09-14 16:44:45,455 - __main__ - INFO - No plate number found for event 1694696816.207402-fnl86q
2023-09-14 16:44:45,456 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694706204.7118-wyd02k', 'camera': 'living_room', 'frame_time': 16947062>
2023-09-14 16:44:45,457 - __main__ - DEBUG - Skipping event: 1694706204.7118-wyd02k because it is from the wrong camera: living_room
2023-09-14 16:44:49,510 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694706204.7118-wyd02k', 'camera': 'living_room', 'frame_time': 16947062>
2023-09-14 16:44:49,510 - __main__ - DEBUG - Skipping event: 1694706204.7118-wyd02k because it is from the wrong camera: living_room
2023-09-14 16:44:57,072 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694706255.752426-p9ig97', 'camera': 'living_room', 'frame_time': 169470>
2023-09-14 16:44:57,072 - __main__ - DEBUG - Skipping event: 1694706255.752426-p9ig97 because it is from the wrong camera: living_room
2023-09-14 16:45:03,318 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694696852.140639-ovlazi', 'camera': 'driveway', 'frame_time': 169470624>
2023-09-14 16:45:03,319 - __main__ - DEBUG - Getting image for event: 1694696852.140639-ovlazi
2023-09-14 16:45:03,319 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694696852.140639-ovlazi/snapshot.jpg
2023-09-14 16:45:04,204 - __main__ - DEBUG - response: {'processing_time': 39.425, 'results': [], 'filename': '1545_5zMQt_upload.jpg', 'version': 1, '>
2023-09-14 16:45:04,205 - __main__ - INFO - No plate number found for event 1694696852.140639-ovlazi
2023-09-14 16:45:11,047 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694706204.7118-wyd02k', 'camera': 'living_room', 'frame_time': 16947062>
2023-09-14 16:45:11,047 - __main__ - DEBUG - Skipping event: 1694706204.7118-wyd02k because it is from the wrong camera: living_room
2023-09-14 16:45:11,481 - __main__ - DEBUG - mqtt message: {'before': {'id': '1694704725.004375-dtzjwd', 'camera': 'front_garden', 'frame_time': 16947>
2023-09-14 16:45:11,482 - __main__ - DEBUG - Getting image for event: 1694704725.004375-dtzjwd
2023-09-14 16:45:11,483 - __main__ - DEBUG - event URL: http://192.168.1.65:5000/api/events/1694704725.004375-dtzjwd/snapshot.jpg
2023-09-14 16:45:12,459 - __main__ - DEBUG - response: {'processing_time': 41.74, 'results': [], 'filename': '1545_3W9nz_upload.jpg', 'version': 1, 'c>
2023-09-14 16:45:12,459 - __main__ - INFO - No plate number found for event 1694704725.004375-dtzjwd `

**Comment by yehpo** [comment]
> can you post the trace stack?

I ran the container for another 5 mins just now ,and here is the log:
```
2023-09-14 16:44:18,316 - __main__ - INFO - Time: 2023-09-14 16:44:18.316
2023-09-14 16:44:18,317 - __main__ - INFO - Python Version: 3.9.18 (main, Sep  7 2023, 13:31:18) 

[GCC 12.2.0]
2023-09-14 16:44:18,317 - __main__ - INFO - Frigate Plate Recognizer Version: 1.3.2
2023-09-14 16:44:18,317 - __main__ - INFO - Starting MQTT client. Connecting to: 192.168.0.201
2023-09-14 16:44:18,320 - __main__ - INFO - MQTT Connected
2023-09-14 16:44:46,883 - __main__ - INFO - Storing plate number in database: ds13wny
2023-09-14 16:44:46,966 - __main__ - INFO - Sublabel set successfully to: ds13wny
2023-09-14 16:44:54,211 - __main__ - INFO - Score is below minimum: 0.772
2023-09-14 16:44:58,653 - __main__ - INFO - No plate number found for event 1694705678.120495-nh0w2s
2023-09-14 16:45:00,886 - __main__ - INFO - Score is below minimum: 0.749
2023-09-14 16:45:27,402 - __main__ - INFO - No plate number found for event 1694693079.113837-777vfv
2023-09-14 16:45:48,542 - __main__ - INFO - No plate number found for event 1694693079.113837-777vfv
2023-09-14 16:45:49,478 - __main__ - INFO - No plate number found for event 1694693079.113837-777vfv
2023-09-14 16:45:54,453 - __main__ - INFO - Score is below minimum: 0.772
2023-09-14 16:45:58,804 - __main__ - INFO - No plate number found for event 1694705678.120495-nh0w2s
2023-09-14 16:46:01,027 - __main__ - INFO - Score is below minimum: 0.749
2023-09-14 16:46:48,648 - __main__ - INFO - No plate number found for event 1694693079.113837-777vfv
2023-09-14 16:46:54,535 - __main__ - INFO - Score is below minimum: 0.772
2023-09-14 16:46:58,927 - __main__ - INFO - No plate number found for event 1694705678.120495-nh0w2s
2023-09-14 16:47:01,170 - __main__ - INFO - Score is below minimum: 0.749
2023-09-14 16:47:48,812 - __main__ - INFO - No plate number found for event 1694693079.113837-777vfv
2023-09-14 16:47:54,641 - __main__ - INFO - Score is below minimum: 0.772
2023-09-14 16:47:59,031 - __main__ - INFO - No plate number found for event 1694705678.120495-nh0w2s
2023-09-14 16:48:01,259 - __main__ - INFO - Score is below minimum: 0.749
```

**Comment by jasonwragg** [comment]
Could it be something in my Frigate config?

**Comment by heisenberg2980** [comment]
@jasonwragg I think so, you need to check the frigate documentation to see why that is happening as Frigate shouldn´t be creating a new event ID unless there is a change (i.e. objects entering/leaving the area)

**Comment by heisenberg2980** [comment]
> > can you post the trace stack?
> 
> I ran the container for another 5 mins just now ,and here is the log:
> 
> ```
> 2023-09-14 16:44:18,316 - __main__ - INFO - Time: 2023-09-14 16:44:18.316
> 2023-09-14 16:44:18,317 - __main__ - INFO - Python Version: 3.9.18 (main, Sep  7 2023, 13:31:18) 
> �
> [GCC 12.2.0]
> 2023-09-14 16:44:18,317 - __main__ - INFO - Frigate Plate Recognizer Version: 1.3.2
> 2023-09-14 16:44:18,317 - __main__ - INFO - Starting MQTT client. Connecting to: 192.168.0.201
> 2023-09-14 16:44:18,320 - __main__ - INFO - MQTT Connected
> 2023-09-14 16:44:46,883 - __main__ - INFO - Storing plate number in database: xxxxxxxx
> 2023-09-14 16:44:46,966 - __main__ - INFO - Sublabel set successfully to: xxxxxxxx
> 2023-09-14 16:44:54,211 - __main__ - INFO - Score is below minimum: 0.772
> 2023-09-14 16:44:58,653 - __main__ - INFO - No plate number found for event 1694705678.120495-nh0w2s
> 2023-09-14 16:45:00,886 - __main__ - INFO - Score is below minimum: 0.749
> 2023-09-14 16:45:27,402 - __main__ - INFO - No plate number found for event 1694693079.113837-777vfv
> 2023-09-14 16:45:48,542 - __main__ - INFO - No plate number found for event 1694693079.113837-777vfv
> 2023-09-14 16:45:49,478 - __main__ - INFO - No plate number found for event 1694693079.113837-777vfv
> 2023-09-14 16:45:54,453 - __main__ - INFO - Score is below minimum: 0.772
> 2023-09-14 16:45:58,804 - __main__ - INFO - No plate number found for event 1694705678.120495-nh0w2s
> 2023-09-14 16:46:01,027 - __main__ - INFO - Score is below minimum: 0.749
> 2023-09-14 16:46:48,648 - __main__ - INFO - No plate number found for event 1694693079.113837-777vfv
> 2023-09-14 16:46:54,535 - __main__ - INFO - Score is below minimum: 0.772
> 2023-09-14 16:46:58,927 - __main__ - INFO - No plate number found for event 1694705678.120495-nh0w2s
> 2023-09-14 16:47:01,170 - __main__ - INFO - Score is below minimum: 0.749
> 2023-09-14 16:47:48,812 - __main__ - INFO - No plate number found for event 1694693079.113837-777vfv
> 2023-09-14 16:47:54,641 - __main__ - INFO - Score is below minimum: 0.772
> 2023-09-14 16:47:59,031 - __main__ - INFO - No plate number found for event 1694705678.120495-nh0w2s
> 2023-09-14 16:48:01,259 - __main__ - INFO - Score is below minimum: 0.749
> ```

In this case it seems the process is alternating between two events and creating an infinite loop, do you have more than one camera configured in the config file?

P.S.: you might want to remove the number plate in the logs you shared here

**Comment by yehpo** [comment]
I do have 2 cameras configured, could that be causing the issue? I can try removing one temporarily to see if that fixes it. 

Regarding the number plates in the logs, I don't see the issue, as they are just random parked cars on the public road, but I will delete them anyway as they are not relevant...

**Comment by heisenberg2980** [comment]
@yehpo check if removing one of the cameras temporarily fixes the issue, but if that is the case the process might still generate infinite loops when more than one camera is setup (@ljmerza FYI)

Regarding the number plate I thought it was from your car, but anyway it is always a good practice to avoid publishing any identifiable information just in case

**Comment by jasonwragg** [comment]
Do you guys have anything in your frigate config to deal with stationary objects?

**Comment by heisenberg2980** [comment]
@ljmerza I think the loop issue is still happening when no number plate is found in the picture:

```
frigate_plate_recognizer  | 2023-09-14 17:12:07,263 - __main__ - INFO - Frigate Plate Recognizer Version: 1.3.2
frigate_plate_recognizer  | 2023-09-14 17:12:07,263 - __main__ - INFO - Starting MQTT client. Connecting to: 192.168.0.253
frigate_plate_recognizer  | 2023-09-14 17:12:07,265 - __main__ - INFO - MQTT Connected
frigate_plate_recognizer  | 2023-09-14 17:27:55,253 - __main__ - INFO - No plate number found for event 1694708868.671571-gjiaz0
frigate_plate_recognizer  | 2023-09-14 17:28:05,130 - __main__ - INFO - No plate number found for event 1694708868.671571-gjiaz0
frigate_plate_recognizer  | 2023-09-14 17:29:05,527 - __main__ - INFO - No plate number found for event 1694708868.671571-gjiaz0
frigate_plate_recognizer  | 2023-09-14 17:30:05,940 - __main__ - INFO - No plate number found for event 1694708868.671571-gjiaz0
frigate_plate_recognizer  | 2023-09-14 17:31:06,385 - __main__ - INFO - No plate number found for event 1694708868.671571-gjiaz0
frigate_plate_recognizer  | 2023-09-14 17:32:06,735 - __main__ - INFO - No plate number found for event 1694708868.671571-gjiaz0
frigate_plate_recognizer  | 2023-09-14 17:33:07,249 - __main__ - INFO - No plate number found for event 1694708868.671571-gjiaz0
```

**Comment by yehpo** [comment]
> @yehpo check if removing one of the cameras temporarily fixes the issue, but if that is the case the process might still generate infinite loops when more than one camera is setup (@ljmerza FYI)

I removed one camera, however, the loop is still happening. 

```
2023-09-14 17:36:55,779 - __main__ - INFO - Time: 2023-09-14 17:36:55.778
2023-09-14 17:36:55,779 - __main__ - INFO - Python Version: 3.9.18 (main, Sep  7 2023, 13:31:18) 

[GCC 12.2.0]
2023-09-14 17:36:55,780 - __main__ - INFO - Frigate Plate Recognizer Version: 1.3.2
2023-09-14 17:36:55,780 - __main__ - INFO - Starting MQTT client. Connecting to: 192.168.0.201
2023-09-14 17:36:55,782 - __main__ - INFO - MQTT Connected
2023-09-14 17:37:12,607 - __main__ - INFO - No plate number found for event 1694709431.019891-sfyw09
2023-09-14 17:37:13,919 - __main__ - INFO - No plate number found for event 1694709242.484646-g20n23
2023-09-14 17:37:14,566 - __main__ - ERROR - Failed to get plate number. Response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-14 17:37:14,566 - __main__ - INFO - No plate number found for event 1694709431.019891-sfyw09
2023-09-14 17:37:15,233 - __main__ - INFO - No plate number found for event 1694709431.019891-sfyw09
2023-09-14 17:37:15,886 - __main__ - ERROR - Failed to get plate number. Response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-14 17:37:15,886 - __main__ - INFO - No plate number found for event 1694709242.484646-g20n23
2023-09-14 17:37:16,574 - __main__ - INFO - No plate number found for event 1694709431.019891-sfyw09
2023-09-14 17:37:17,195 - __main__ - ERROR - Failed to get plate number. Response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-14 17:37:17,195 - __main__ - INFO - No plate number found for event 1694709431.019891-sfyw09
2023-09-14 17:37:17,855 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:18,498 - __main__ - ERROR - Failed to get plate number. Response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-14 17:37:18,498 - __main__ - INFO - No plate number found for event 1694709431.019891-sfyw09
2023-09-14 17:37:19,187 - __main__ - INFO - Score is below minimum: 0.677
2023-09-14 17:37:19,822 - __main__ - ERROR - Failed to get plate number. Response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-14 17:37:19,823 - __main__ - INFO - No plate number found for event 1694709431.019891-sfyw09
2023-09-14 17:37:23,850 - __main__ - INFO - No plate number found for event 1694709431.019891-sfyw09
2023-09-14 17:37:27,422 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:28,687 - __main__ - INFO - No plate number found for event 1694709242.484646-g20n23
2023-09-14 17:37:30,638 - __main__ - INFO - No plate number found for event 1694709431.019891-sfyw09
2023-09-14 17:37:32,471 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:46,718 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:47,334 - __main__ - ERROR - Failed to get plate number. Response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-14 17:37:47,334 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:47,993 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:48,632 - __main__ - ERROR - Failed to get plate number. Response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-14 17:37:48,633 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:49,283 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:49,912 - __main__ - ERROR - Failed to get plate number. Response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-14 17:37:49,912 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:50,607 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:51,238 - __main__ - ERROR - Failed to get plate number. Response: {'detail': 'Request was throttled. Expected available in 1 second.', 'status_code': 429}
2023-09-14 17:37:51,238 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:37:51,891 - __main__ - INFO - No plate number found for event 1694709467.591859-h99kal
2023-09-14 17:37:54,920 - __main__ - INFO - No plate number found for event 1694709434.886798-uys6du
2023-09-14 17:38:00,177 - __main__ - INFO - No plate number found for event 1694709467.591859-h99kal
2023-09-14 17:39:00,474 - __main__ - INFO - No plate number found for event 1694709467.591859-h99kal
2023-09-14 17:39:44,944 - __main__ - INFO - No plate number found for event 1694709467.591859-h99kal
```

**Comment by ljmerza** [comment]
I would use mqtt explorer to see if these events are being published again and again to mqtt.

**Comment by yehpo** [comment]
> I would use mqtt explorer to see if these events are being published again and again to mqtt.

It looks like it is, however, it using the same event ID. Is this not the expected behaviour?

**Comment by heisenberg2980** [comment]
I think the issue is not due to multiple cameras, but due to this process not marking the event as complete when plate recognizer cannot find any number plate in the picture.

@ljmerza the events are being published again and again but that is expected, the process should mark the event as complete in both cases regarthless of plate recognizer finding a plate number or not, otherwise the loop happens

**Comment by ljmerza** [comment]
I don't think the event should be published again and again but I can add to db to stop this.

**Comment by ljmerza** [comment]
`frigate/events`

**Comment by yehpo** [comment]
that topic lists all current events anytime a new event starts or ends, I believe. In which case you will receive multiple messages regarding the same event...

**Comment by yehpo** [comment]
there are 3 types of messages, new, update and end

maybe we should only track the new messsages?

**Comment by yehpo** [comment]
> that topic lists all current events anytime a new event starts or ends, I believe. In which case you will receive multiple messages regarding the same event...

this is actually incorrect, but the same event can be published multiple times, as above.

**Comment by heisenberg2980** [comment]
> there are 3 types of messages, new, update and end
> 
> maybe we should only track the new messsages?

That´s a good idea as well. I guess solution 1 (processing only one snapshot per event ID by updating the status to "processed") will produce the same result, so whatever solution is easier to implement. The only situation where processing only the "new" messages would be better is in case the process cannot update the status of the event to "processed" for any reason, in that case the solution 1 would still produce a loop while this solution 2 would not.

**Comment by heisenberg2980** [comment]
@ljmerza what is your view about these 2 potential solutions? I think once one of these solutions is implemented we should be able to test it and close this issue.

**Comment by ljmerza** [comment]
I assume update events are the same object but potentially better detections. If we only process new events we are missing out on potentially better matches. For example, jf a car is driving by and the new event is the front of the car but then update events are as the car is passing by you'll miss the license plate for every event.

**Comment by heisenberg2980** [comment]
You are right, ideally we would want to process every picture if they have not been processed yet, but avoiding sending the same picture more than once. Not sure if there is a way for the process to know if the exact picture has already been sent and call plate recognizer only if the picture is new, but if that is difficult or not possible, avoiding the infinite loop should be our first priority to avoid people wasting the 2500 picture allowance with duplicate pictures

**Comment by heisenberg2980** [comment]
@ljmerza I might have found an easier way to fix this loop issue while still processing better pictures for the same event id. Looking at the mqtt message generated by frigate I realised the events topic include before and after data, and both of them have a field called top_score, which I guess is the field that frigate uses to decide if a new snapshot needs to be generated for that event id (i.e. when the new top score is higher than the previous one), so if that is the case the process could just compare those two values and process the new snapshot only if the "after" top_score is higher than the "before" top_score.

**Comment by yehpo** [comment]
It would also be nice if we can limit the process to specific zones, not only cameras, that way we can skip the detection for zones where the plate wouldn't be visible.

**Comment by heisenberg2980** [comment]
> It would also be nice if we can limit the process to specific zones, not only cameras, that way we can skip the detection for zones where the plate wouldn't be visible.

Please create a new request for that as it is not related to this issue

**Comment by yehpo** [comment]
> I think what he is saying is frigate is detecting a car so the score in relation to if a license plate is viewable or not is irrelevant

Exactly. In fact, a car is more recognisable from the side, so it is entirely possible that Frigate may prefer side photos, although that is an uninformed assumption!

**Comment by heisenberg2980** [comment]
@ljmerza the top score is relevant if that is what frigate uses to update the snapshot (I haven´t found it in the frigate documentation regarding this, but that is what I believe with the test I have done). Using that top score we will make sure that we only send images to plate recognizer when frigate has generate a new snapshot and we will never send duplicates

@yehpo what I am trying to say is that frigate is the one generating the snapshots, so if frigate doesn´t consider the new picture has a higher score it will not update the snapshot (no matter if the number plater is more visible). The only time that the process can send a new snapshot to plate recognizer is when frigate updates the snapshot because it considers it is a better one, and I believe frigate uses the score to do that

**Comment by yehpo** [comment]
I understand you now.

**Comment by heisenberg2980** [comment]
@ljmerza I have created this PR that I believe it should fix the loop issue: https://github.com/ljmerza/frigate_plate_recognizer/pull/6

It´s a prety simple change but I haven´t been able to test it yet (I still need to figure out how to test changes locally), please feel free to review it and merge it if you are happy with it

**Comment by ljmerza** [comment]
updated docker container with the PR

**Comment by heisenberg2980** [comment]
Thanks @ljmerza 

@jasonwragg if the loop is not happening anymore you can close this issue
